aws/solutions/ManagedAD/templates/MANAGEDAD.cfn.yaml

```diff
@@ -0,0 +1,398 @@
+        default: Create AWS resources to support seamless domain join Windows EC2 instances
+      PrivateSubnet1ID:
+        default: Private Subnet 1 ID
+      PrivateSubnet2ID:
+        default: Private Subnet 2 ID
+      SecretsManagerDomainCredentialsSecretsKMSKey:
+        default: Secrets Manager KMS Key for domain credentials secret
+      SSMLogsBucketName:
+        default: Systems Manager (SSM) Logs Bucket Name
+      VPCID:
+        default: VPC ID
+Parameters:
+  AWSManagedADDomainDNSName:
+    AllowedPattern: '[a-zA-Z0-9-]+..+'
+    Default: awsmad.lab
+    Description: Fully qualified domain name for the AWS Managed Microsoft AD directory, such as corp.example.com.
+    Type: String
+  AWSManagedADDomainNetBiosName:
+    AllowedPattern: '[a-zA-Z0-9\-]+'
+    Default: AWSMAD
+    Description: NetBIOS name for your domain, such as CORP.
+    MaxLength: 15
+    MinLength: 1
+    Type: String
+  AWSManagedADAdminPassword:
+    AllowedPattern: '(?=^.{6,255}$)((?=.*\d)(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[^A-Za-z0-9])(?=.*[a-z])|(?=.*[^A-Za-z0-9])(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[A-Z])(?=.*[^A-Za-z0-9]))^.*'
+    Description:
+      Password for the default administrative user named Admin. 8-64 characters, and must match at least 3 of the 4. (1) lowercase letters (2)
+      Uppercase letters (3) numbers (4) non-alphanumeric characters "~!@#$%^&*_-+=`|\(){}[]:;"'<>,.?/"
+    NoEcho: true
+    Type: String
+  AWSManagedADEdition:
+    AllowedValues: [Standard, Enterprise]
+    Default: Standard
+    Description: AWS Managed AD Edition. Standard supports up to 30,000+ directory objects. Enterprise supports up to 500,000+ directory objects.
+    Type: String
+  CreateDHCPOptionSet:
+    AllowedValues: ['Yes', 'No']
+    Default: 'No'
+    Description: Create DHCP Option Set
+    Type: String
+  CreateDomainMembersSG:
+    AllowedValues: ['Yes', 'No']
+    Default: 'No'
+    Description: Create Domain Members Security Group. Note, using allow any type rules, restrict accordingly.
+    Type: String
+  CreateLinuxEC2DomainJoinResources:
+    AllowedValues: ['Yes', 'No']
+    Default: 'No'
+    Description: Create AWS resources (IAM role, instance profile, & secret) to support seamless domain join Linux EC2 instances
+    Type: String
+  CreateWindowsEC2DomainJoinResources:
+    AllowedValues: ['Yes', 'No']
+    Default: 'No'
+    Description: Create AWS resources (IAM role & instnace profile)to support seamless domain join Windows EC2 instances
+    Type: String
+  PrivateSubnet1ID:
+    Description: ID of the private subnet 1 in Availability Zone 1 (e.g., subnet-a0246dcd)
+    Type: AWS::EC2::Subnet::Id
+  PrivateSubnet2ID:
+    Description: ID of the private subnet 2 in Availability Zone 2 (e.g., subnet-a0246dcd)
+    Type: AWS::EC2::Subnet::Id
+  SecretsManagerDomainCredentialsSecretsKMSKey:
+    AllowedPattern: '^$|^arn:(aws[a-zA-Z-]*)?:kms:[a-z0-9-]+:\d{12}:key\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$'
```

Comment:
Should the first group here (the partition) be optional?

Reply:
it should not, I will correct it.